client/state/http/test/index.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import sinon from 'sinon';
+
+/**
+ * Internal dependencies
+ */
+import useMockery from 'test/helpers/use-mockery';
+import { extendAction } from 'state/utils';
+import { failureMeta, successMeta } from 'state/data-layer/wpcom-http';
+
+const succeeder = { type: 'SUCCESS' };
+const failer = { type: 'FAIL' };
+
+const requestDomain = 'https://api.some-domain.com:443';
+const requestPath = '/endpoint';
+const getMe = {
+	url: requestDomain + requestPath,
+	method: 'GET',
+	onFailure: failer,
+	onSuccess: succeeder,
+};
+
+class SuperAgentMock {
+	constructor() {
+		this._success = false;
+		this._data = new Error( 'Mock response is not set' );
+		this._lastRequest = null;
+	}
+
+	/***
+	 * Mocks superagent() call that returns request
+	 *
+	 * const request = superagent( method, url );
+	 *
+	 * @param {String} method HTTP method
+	 * @param {String} url URL
+	 * @returns {Object} superagent like request with sinon spies
+	 */
+	handler = ( method, url ) => {
+		const request = {
+			method,
+			url,
+			withCredentials: sinon.spy(),
+			set: sinon.spy(),
+			accept: sinon.spy(),
+			send: sinon.spy(),
+			then: ( successHandler, failureHandler ) =>
+				this._success
+					? successHandler( { body: this._data } )
+					: failureHandler( { response: { body: this._data } } ),
+		};
+
+		this._lastRequest = request;
+
+		return request;
+	};
+
+	/***
+	 * Tells the mock how to act, successful response or failure
+	 *
+	 * @param {Boolean} success should the request call success handler or failure?
+	 * @param {Object} data the data that should be passed in body
+	 */
+	setResponse( success, data ) {
+		this._success = success;
+		this._data = data;
+	}
+
+	/***
+	 * Get last mocked request
+	 * @returns {Object} An object similar to superagent's request with additional `method` and `url` fields
+	 */
+	getLastRequest() {
+		return this._lastRequest;
+	}
+}
+
+const superagentMock = new SuperAgentMock();
+
+describe( '#httpHandler', () => {
+	let next;
+	let dispatch;
+	let httpHandler;
+
+	useMockery( mockery => mockery.registerMock( 'superagent', superagentMock.handler ) );
+
+	before( () => {
+		httpHandler = require( '../' ).httpHandler; // it's done here for mockery to work
+	} );
+
+	beforeEach( () => {
+		dispatch = sinon.spy();
+		next = sinon.spy();
+	} );
+
+	it( 'should call `onSuccess` when a response returns with data', () => {
+		const data = { value: 1 };
+
+		superagentMock.setResponse( true, data );
+
+		httpHandler( { dispatch }, getMe, next );
+
+		expect( dispatch ).to.have.been.calledOnce;
+
+		sinon.assert.calledWith(
+			dispatch,
+			sinon.match(
+				extendAction(
+					succeeder,
+					successMeta( { body: data } )
+				)
+			)
+		);
```

Comment:
I'm always curious about this when I see people prefer assertions like this (the family of `.assert.XYZ` functions) over `expect()`. Are you willing to share what you like better about it?